include/deal.II/base/mpi.templates.h

```diff
@@ -324,6 +324,71 @@ namespace Utilities
 
 
 
+    template<typename T>
+    std::vector<T> gather(const MPI_Comm &comm,
+                          const T &object)
+    {
+#ifndef DEAL_II_WITH_MPI
+      (void)comm;
+      std::vector<T> v(1, object);
+      return v;
+#else
+      const auto n_procs = dealii::Utilities::MPI::n_mpi_processes(comm);
+      const auto my_rank = dealii::Utilities::MPI::this_mpi_process(comm);
+
+      std::vector<char> buffer = Utilities::pack(object);
+      int n_local_data = buffer.size();
+
+      // Vector to store the size of loc_data_array for every process
+      // only the root process needs to allocate memory for that purpose
+      std::vector<int> size_all_data;
+      if (my_rank==0)
+        size_all_data.resize(n_procs,0);
+
+      // Exchanging the size of each buffer
+      MPI_Gather(&n_local_data, 1, MPI_INT,
+                 size_all_data.data(), 1, MPI_INT,
+                 0,comm);
```

Comment:
maybe instead of hard-coding `0`, add 
```
const unsigned int root = 0;
```
at the top and use it throughout.

Reply:
Actually, I would prefer if the id of the receiving MPI process is a parameter (with default value 0).